tests/integration/telemetry/api/testdata/additional-labels.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: additional-labels
+  namespace: istio-system
+spec:
+  metrics:
+    - overrides:
+        - match:
+            metric: ALL_METRICS
+            mode: CLIENT
+          tagOverrides:
+            custom_label:
+              value: filter_state.upstream_peer.labels['custom-label']
```

Comment:
:thinking: how does this pass? don't we need PILOT_MX_ADDITIONAL_LABELS=true set on the istiod install for the test?

Reply:
good catch, I tought I did.
it's odd that tests didn't failed.

Reply:
```
--- PASS: TestStatsFilter (2.13s)
    --- PASS: TestStatsFilter/default (2.07s)
    --- PASS: TestStatsFilter/additional-labels (0.04s)
```

the test passed too quickly, It doesn't look like there's running actually.

Reply:
`custom_label` tag was overrided when prometheus scraped the metrics endpoint, because of same tag name.